### PR TITLE
network_engine terminology

### PR DESCRIPTION
**Pull request by trishnaguha**
**SUMMARY:**

Fixes https://github.com/ansible-network/network-engine/issues/65

- Remove action plugin code from `text_parser`, `textfsm` and load newly added action plugins `command_parser` and `textfsm_parser` with `action_loader` to support backward compatibility and avoid repetition of code.
- Update `cli` action plugin
- Fix module doc
- Add new tests
- Add deprecation message
- Fix docs/

PASSES all tests

### CODE REVIEW

**Comment by privateip** [comment]
Lets be sure that we fully vet out https://github.com/ansible-network/network-engine/issues/65 before this gets merged.

**Review by privateip — APPROVED** [review]

File: library/_text_parser.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/python
```
Comment (acozine) [inline library/_text_parser.py:1]: This file should be renamed `library/_command_parser.py`

Comment (acozine) [reply library/_text_parser.py:1]: Oh, wait, is this the backwards-compatibility part? If so, ignore my comment above.

File: library/_textfsm.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
```
Comment (acozine) [inline library/_textfsm.py:1]: Same thing here. If this is for backwards compatibility, it's good. If it's the working code, then it should be renamed `library/_textfsm_parser.py`

File: tests/command_parser/parsers/ios/show_interfaces.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+---
```
Comment (acozine) [inline tests/command_parser/parsers/ios/show_interfaces.yaml]: The directory should be renamed `tests/command_parser/parser_templates/`

File: tests/textfsm_parser/parsers/ios/show_interfaces [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Value Required name (\S+)
```
Comment (acozine) [inline tests/textfsm_parser/parsers/ios/show_interfaces]: Same thing here: `tests/textfsm_parser/parser_templates/`

**Pull request merged**